.github/actions/build-test/action.yml

```diff
@@ -0,0 +1,101 @@
+# This action build and tests the binary. The Conan dependencies must have
+# already been installed (see the build-deps action).
+name: Build and Test
+
+inputs:
+  build_dir:
+    description: 'The directory where to build.'
+    required: true
+    type: string
+  build_type:
+    description: 'The build type to use.'
+    required: true
+    type: string
+  cmake_args:
+    description: 'Additional arguments to pass to CMake.'
+    required: false
+    type: string
+    default: ''
+  cmake_generator:
+    description: 'The CMake generator to use for the build.'
+    required: true
+    type: string
+  cmake_target:
+    description: 'The CMake target to build.'
+    required: true
+    type: string
+  codecov_token:
+    description: 'The Codecov token to use for uploading coverage reports.'
+    required: false
+    type: string
+    default: ''
+  os:
+    description: 'The operating system that is used (Linux, MacOS, or Windows).'
+    required: true
+    type: choice
+    options:
+      - Linux
+      - MacOS
+      - Windows
+
+runs:
+  using: composite
+  steps:
+    - name: Configure CMake
+      shell: bash
+      working-directory: ${{ inputs.build_dir }}
+      run: |
+        echo 'Configuring CMake.'
+        cmake \
+          ${{ inputs.cmake_generator && format('-G "{0}"', inputs.cmake_generator) || '' }} \
```

Comment:
Do we need this long expression here ? It seems like we always set `cmake_generator`